### PR TITLE
Admission check endpoint

### LINKED ISSUE #336: Unable to easily verify if the npub has paid the invoice
**Is your feature request related to a problem? Please describe.**
We have an external application that makes use of a dedicated relay (noStream), and in order to facilitate the login process, we need a mechanism by which we can find out if the user is an active (paid) user on the relay 

**Describe the solution you'd like**
A `GET` request on URL such as `/invoices/check/:pubkey` against which the check will be done and returned in a response of a JSON type.

**Describe alternatives you've considered**
The alternative I tried was sending a `POST` request every 20 seconds to the `/invoice` endpoint which breaks the relay. It fails to process successful payment after that kind of polling.

**Additional Context**
We're currently preparing a PR for submission.

### PR DESCRIPTION
**Pull request by medjedovicm**
## Description
Created `admission router` and `admission check controller` which runs a check against the provided `pubkey` in the URL parameter and returns JSON with information if the user (pubkey) is admitted or not.

`GET` `/admissions/check/:pubkey`

Response:
```
{
  "userAdmitted": boolean
}
```

## Related Issue
Closes #336 

## Motivation and Context
In our use case, we need to check if a user is admitted on our own relay before we let them log in.
The user is redirected to the relay page (in the new tab) while we send a check request every 10 seconds or so, so we can automatically figure out when payment is done.

## How Has This Been Tested?
Visited a new endpoint in the browser, looking for the response.
As far as I am aware my changes do not affect any other parts of the code.

## Screenshots (if appropriate):
image

## Types of changes
- [x] Non-functional change (docs, style, minor refactor)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my code changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/controllers/invoices/get-invoice-pubkey-check-controller.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        .status(429)
+        .setHeader('content-type', 'text/plain; charset=utf8')
+        .send('Too many requests')
+      return
+    }
+    
+    const pubkey = request.params.pubkey
+    const user = await this.userRepository.findByPubkey(pubkey)
+
+    console.log('user', user)
```
Comment (Cameri) [inline src/controllers/invoices/get-invoice-pubkey-check-controller.ts:34]: Please remove

File: src/controllers/invoices/get-invoice-pubkey-check-controller.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    response
+      .status(200)
+      .setHeader('content-type', 'application/json; charset=utf8')
+      .send({ userAdmitted })
+
+    return
+  }
+
+  public async isRateLimited(request: Request, settings: Settings) {
+    const rateLimits = path(['limits', 'invoice', 'rateLimits'], settings)
```
Comment (Cameri) [inline src/controllers/invoices/get-invoice-pubkey-check-controller.ts:52]: We should have a different rate limit for this endpoint

**Review by Cameri — APPROVED** [review]

**Comment by medjedovicm** [comment]
I'm sorry, I missed a crucial commit... I think this structure makes more sense. Another commit is coming with an addressed rate limits comment.

**Comment by Cameri** [comment]
@medjedovicm ive fixed the integration tests, do you mind rebasing off the main branch to incorporate the fix into yours?

**Comment by medjedovicm** [comment]
@Cameri Yes sure, done.

**Review by Cameri — APPROVED** [review]

**Pull request merged**